hyperspy/model.py

```diff
@@ -126,11 +126,18 @@ def reconstruct_component(comp_dictionary, **init_args):
         # it is serialized using dill.
         _class = dill.loads(comp_dictionary['_class_dump'])
     else:
-        raise ImportError(
-            f'Loading the {comp_dictionary["class"]} component ' +
-            'failed because the component is provided by the ' +
-            f'{comp_dictionary["package"]} Python package, but ' +
-            f'{comp_dictionary["package"]} is not installed.')
+        try:
+            raise ImportError(
+                f'Loading the {comp_dictionary["class"]} component ' +
+                'failed because the component is provided by the ' +
+                f'{comp_dictionary["package"]} Python package, but ' +
+                f'{comp_dictionary["package"]} is not installed.')
+        except KeyError:
```

Comment:
Would it be possible to make this more precise e.g. by checking if the component's package is HyperSpy and `comp_dictionary["class"]` is one of:

```python
EXSPY_HSPY_COMPONENTS = ("EELSArctan", "EELSCLEdge", "DoublePowerLaw", "Vignetting", "PESCoreLineShape", "SEE", "PESVoigt", "VolumePlasmonDrude")
```

?